src/components/AccountControl.tsx

```diff
@@ -1,36 +1,28 @@
-
-import React, { useEffect } from 'react';
+import React from 'react';
 import { AccountInfo } from './AccountInfo';
-import { useWeb3React } from '@web3-react/core';
-import { Injected } from '../utils/web3';
+import useWeb3Modal from "../hooks/useWeb3Modal";
+import useUserAddress from '../hooks/useUserAddress';
 
 export const AccountControl = () => {
-    const context = useWeb3React();
-
-    useEffect(() => {
-        async function asyncEffect() {
-            const authorized = await Injected.isAuthorized();
-
-            if (authorized) { 
-                await connectWeb3();
-            }
-        }
-        
-        asyncEffect();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
-
-    async function connectWeb3() {
-        try { 
-            await context.activate(Injected, undefined, true);
-        } catch (error) {
-            console.log(error)
-        }
-    }
+  const [
+    provider,
+    loadWeb3Modal,
+    logoutOfWeb3Modal,
+	] = useWeb3Modal();
+	const [account] = useUserAddress(provider);
 
-    if (context.account) { 
-        return <AccountInfo address={context.account} /> 
+	const onClick = () => {
+    if (!provider) {
+      loadWeb3Modal();
+    } else {
+      logoutOfWeb3Modal();
     }
+  }
+  if (account) {
+      return <AccountInfo address={account} />
+  }
 
-    return <button type="button" className="btn btn-outline-info btn-sm" onClick={connectWeb3}>Connect</button>
-}
\ No newline at end of file
+  return <button type="button" className="btn btn-outline-info btn-sm" onClick={onClick}>
+    {!provider ? "Connect Wallet" : "Disconnect Wallet"}
```

Comment:
I think that 'Disconnect Wallet' would never get shown. 
You either have a provider, because you've logged in. Which shows your account info. 
Or you're not logged in and it shows a connect button.